### PR TITLE
Add error message and better error checking

### PR DESCRIPTION
**Pull request by yuzhu**
@ns1123 I will add a test case after this.

### CODE REVIEW

**Comment by ns1123** [comment]
This is a breaking change for any user using tiered UFS currently. Can we assume medium type as MEM for tier0, SSD for tier1, and HDD for tier2 and higher?

File: core/server/worker/src/main/java/alluxio/worker/block/meta/StorageTier.java [hunk 1]
```diff
@@ -82,6 +82,7 @@ private void initStorageTier()
     PropertyKey tierDirMediumConf =
         PropertyKey.Template.WORKER_TIERED_STORE_LEVEL_DIRS_MEDIUMTYPE.format(mTierOrdinal);
     String rawDirMedium = ServerConfiguration.get(tierDirMediumConf);
+    Preconditions.checkState(rawDirMedium.length() > 0, PreconditionMessage.ERR_TIER_MEDIUM_TYPE_BLANK);
```
Comment (apc999) [inline core/server/worker/src/main/java/alluxio/worker/block/meta/StorageTier.java]: I would suggest not to use `PreconditionMessage`. It turned out a lot of issues in debugging

**Comment by yuzhu** [comment]
@ns1123 The default value for medium type ais MEM for tier0, SSD for tier1, and HDD for tier2. We don't have a default for any tier higher than 2. But the code will now handle that and give a warning.

**Comment by yuzhu** [comment]
@ns1123 added a test case, PTAL. 
thanks.

**Comment by yuzhu** [comment]
jenkins, test this please

**Comment by ns1123** [comment]
LGTM

**Comment by yuzhu** [comment]
@calvinjia PTAL and merge. Thanks!

**Comment by calvinjia** [comment]
alluxio-bot, merge this please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Pull request merged**